### PR TITLE
Implement enemy behaviour framework

### LINKED ISSUE #25: Implement enemy framework
We need a framework for enemies and creating them that is modular and takes into account future features.

### PR DESCRIPTION
**Pull request by Sirkular**
Implemented a framework for our enemy AI. 
The expected flow:
1. The spawning mechanism utilizes EnemyFactory by passing a name of a prefab (enemy) and position to EnemyFactory.CreateEnemy(). TODO: Currently targeting, moving, and pathfinding is defaulted, but we have the ability to dynamically change it easily. We need to determine how we want to set those up.
2. EnemyFactory instantiates the prefab and sets its location, then calls EnemyBehaviour.AddToGameObject() with the dynamic modules (pathfinding, moving, aiming).
3. EnemyBehaviour.AddToGameObject() adds the EnemyBehaviour script as a component to the prefab/game object, then initializes the EnemyBehaviour with the dynamic module arguments.
4. EnemyFactory calls additional setup from _enemySetups if any exist or are required.

Closes #25

### CODE REVIEW

**Comment by MaXeraph** [comment]
Maybe make a ticket for that TODO in the description if you haven't done it

**Comment by Sirkular** [comment]
> Maybe make a ticket for that TODO in the description if you haven't done it

Should be done when we work on #5 or #6, which is why I didn't separate it out into another issue.

File: Assets/Scripts/Enemy/EnemyBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    // Transform of the enemy's target.
+    private Transform _targetTransform;
+    private Quaternion _lookRotation;
+    private Vector3 _direction;
+    private bool _cooldown = false;
+    private float _cooldownDelay = 1f;
+    // _weapon = new Weapon();
+
+    // Method to pathfind from Transform to Transform, returns Vector3, the direction to move in next.
+    private Func<Transform, Transform, Vector3> Pathfind;
```
Comment (MaXeraph) [inline Assets/Scripts/Enemy/EnemyBehaviour.cs:21]: Is this `pathFind` utilizing the mesh pathfinding solution that @ethanakester came across ?

**Review by MaXeraph — APPROVED** [review]
Otherwise it looks good.

File: Assets/Scripts/Enemy/EnemyBehaviour.cs (hunk 1 above)
Comment (Sirkular) [reply Assets/Scripts/Enemy/EnemyBehaviour.cs:21]: A pathfinding algorithm doesn't exist yet. This is only the framework to support a solution, the pathfinding solution should be implemented week 2 IIRC.

Comment (Toe-Knee-Y) [reply Assets/Scripts/Enemy/EnemyBehaviour.cs:21]: We should discuss the pathfinding sometime this week as we have an idea of which direction we want to take. (either using Ethan's method or something else)

**Review by Toe-Knee-Y — APPROVED** [review]

**Comment by Toe-Knee-Y** [comment]
Looks good, just need to discuss the pathfinding sometime this week

**Review by ethanakester — APPROVED** [review]

**Pull request merged**